src/message.rs

```diff
@@ -305,7 +305,7 @@ impl Message {
             }) => {
                 vec![
                     join_reference.into(),
-                    reference.as_ref().map(From::from).unwrap_or(Value::Null),
+                    reference.as_ref().map(|reference| serde_json::to_value(reference).unwrap()).unwrap_or(Value::Null),
```

Comment:
use and_then instead of map so that this closure can take an option, then use ok to convert Result into Option